### DISCUSSION TITLE
Installation of openDTU on a Espressiv ESP32-C3-DevKit C-02?

### DISCUSSION THREAD

**Discussion by hpunkts** [tbnobody/OpenDTU#735] [Q&A]
I cannot find the lates stable version of OpenDTU.
Is it Actions-> OpenDTU-> MASTER or pull or push?
I see push and pulls, but not a single stable *.bin version.

In Chrome ESPTool, i program the 4 bin files and receive this:
esptool.js
Serial port WebSerial VendorID 0x10c4 ProductID 0xea60
Connecting....
Detecting chip type... ESP32-C3
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 58:cf:79:04:40:e0
Uploading stub...
Running stub...
Stub running...
Changing baudrate to 115200
Changed
Compressed 17408 bytes to 12112...
Writing at 0x1000... (100%)
Wrote 17408 bytes (12112 compressed) at 0x1000 in 1.527 seconds.
Hash of data verified.
Compressed 3072 bytes to 129...
Writing at 0x8000... (100%)
Wrote 3072 bytes (129 compressed) at 0x8000 in 0.11 seconds.
Hash of data verified.
Compressed 8192 bytes to 47...
Writing at 0xe000... (100%)
Wrote 8192 bytes (47 compressed) at 0xe000 in 0.193 seconds.
Hash of data verified.
Compressed 1485776 bytes to 975323...
Writing at 0x10000... (1%)
Writing at 0x14bbb... (3%)
Writing at 0x18b92... (5%)
Writing at 0x1ce68... (6%)
Writing at 0x20e63... (8%)
Writing at 0x24e5e... (10%)
Writing at 0x28e59... (11%)
Writing at 0x2ce54... (13%)
Writing at 0x30e4f... (15%)
Writing at 0x34e4a... (16%)
Writing at 0x3b589... (18%)
Writing at 0x4531b... (20%)
Writing at 0x53048... (21%)
Writing at 0x5c828... (23%)
Writing at 0x671d5... (25%)
Writing at 0x799b6... (26%)
Writing at 0x7f374... (28%)
Writing at 0x8532f... (30%)
Writing at 0x8bc54... (31%)
Writing at 0x921e8... (33%)
Writing at 0x9913a... (35%)
Writing at 0xa033f... (36%)
Writing at 0xa617d... (38%)
Writing at 0xac128... (40%)
Writing at 0xb226e... (41%)
Writing at 0xb8312... (43%)
Writing at 0xbe101... (45%)
Writing at 0xc3b2d... (46%)
Writing at 0xc95ea... (48%)
Writing at 0xcfe94... (50%)
Writing at 0xd5335... (51%)
Writing at 0xda456... (53%)
Writing at 0xdf7f3... (55%)
Writing at 0xe4b43... (56%)
Writing at 0xe9ebf... (58%)
Writing at 0xef384... (60%)
Writing at 0xf43ab... (61%)
Writing at 0xf995e... (63%)
Writing at 0xfeaf5... (65%)
Writing at 0x103c6f... (66%)
Writing at 0x1095a2... (68%)
Writing at 0x10ed24... (70%)
Writing at 0x1143d2... (71%)
Writing at 0x1198b3... (73%)
Writing at 0x11ea3e... (75%)
Writing at 0x123e46... (76%)
Writing at 0x128fdd... (78%)
Writing at 0x12e69a... (80%)
Writing at 0x133ed8... (81%)
Writing at 0x139ba9... (83%)
Writing at 0x13f636... (85%)
Writing at 0x144fbc... (86%)
Writing at 0x14a8ed... (88%)
Writing at 0x150743... (90%)
Writing at 0x1599e6... (91%)
Writing at 0x1622cc... (93%)
Writing at 0x1677ba... (95%)
Writing at 0x16d114... (96%)
Writing at 0x1722fa... (98%)
Writing at 0x177a30... (100%)
Wrote 1485776 bytes (975323 compressed) at 0x10000 in 88.011 seconds.
Hash of data verified.
Leaving...

Then, I push each of 2 buttons (boot, reset), but the red light is still on.
Maybe, I cannot reset.
Then, I connect a USB power supply, and wait for Wifi SSID - no Wifi comes up.

Maybe I have to buy official openDTU as a hardware...
Do I operate wrong ESP32 chip?

Any idea?

**Comment by Solarteur** [comment]
First of all: No "official" OpenDTU hardware is sold! The necessary hardware is described here: https://github.com/tbnobody/OpenDTU#hardware-you-need.

Regarding the firmware: You can find the latest stable version at:
Actions --> OpenDTU Build
Just choose the latest build from the master branch (search for "master" in the blue font text but click on the white header text!).

But: These pre-compiled.bin files are intended for an ESP32 board with 38 pins (see https://github.com/tbnobody/OpenDTU#esp32-board). Maybe your board is supported, but not with the pre-compiled .bin files, as far as I can tell. You have to check this in Visual Studio Code and compile the .bin yourself if necessary.

**Reply by hpunkts** [reply]
OK, Thank you.
the print on my chip shows "ESP32 C3 WROOM 02"
Google chrome and ESPTool shows: "ESP32-C3"
Serial port WebSerial VendorID 0x10c4 ProductID 0xea60
Connecting....
Detecting chip type... ESP32-C3
Chip is ESP32-C3 (revision 3)
Features: Wi-Fi
Crystal is 40MHz
MAC: 58:cf:79:04:40:e0
Uploading stub...
Running stub...
Stub running...
Changing baudrate to 115200
Changed

In ArduinoIDE, I am able to chose bord "ESP32C3 Dev Module" and the blink test program shows a blinking builtin LED!
Using board 'esp32c3' from platform in folder: /Users/cc/Library/Arduino15/packages/esp32/hardware/esp32/2.0.7
Using core 'esp32' from platform in folder: /Users/cc/Library/Arduino15/packages/esp32/hardware/esp32/2.0.7
Result:
Writing at 0x00049422... (100 %)
Wrote 242816 bytes (136035 compressed) at 0x00010000 in 12.1 seconds (effective 160.4 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

In VSCode, I failed to run blink, because only 2 other boards were offered 
a) "esp32" and 
b) "esp32-c3-devkitm-1".

a) is the wrong title and 
b) showed as a result:
 *  Task wird ausgeführt: platformio run --target upload 
Processing generic (board: esp32-c3-devkitm-1; framework: arduino; platform: espressif32@>=6.0.1)
-----------
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)
Tool Manager: Could not parse manifest -> Expecting value: line 1 column 1 (char 0)

kann man hier auch auf deutsch schreiben?

Das sind meine Optionen:
+neue HW kaufen und per *.bin und Chrome ESPTool flashen
+einen Programmierer einstellen?
+die DTUpro für `200,--€ kaufen?

Zu 2 oben: auf der Seite github openDTU habe ich nur die jeweils letzten fixe gefunden und keine vollständigen Builds.
Ich habe es dadurch gelöst, dass ich in VSCode die Adresse gecloned habe und dann nichts auswählen musste. Offensichtlich sucht sich dann VScode das main.cpp ...

Viele Grüße,

**Comment by Solarteur** [comment]
Man kann hier auch auf Deutsch schreiben - dafür gibt es genügend Beispiele...

Zu kompletten Builds: https://github.com/tbnobody/OpenDTU/actions/runs/4471054882 ist der letzte verfügbare vollständige Build - aber so nur für das hier https://github.com/tbnobody/OpenDTU#hardware-you-need empfohlene Board verwendbar. Das klappt problemlos.

In der platformio.ini sind neben [env:generic] - das oben empfohlene Board, weitere ESP32-Boards "angelegt". Ich habe aber keines gefunden, was zum ESP32-C3-DevKit C-02 passt.

**Comment by fhempy** [comment]
Ich habe ein ESP32-C3-13U Board und gesehen, dass es in der platform.io ein env:generic_esp32c3 gibt. Build und Upload haben super funktioniert und auch der AP Wifi ist verfügbar.
Einziges Problem, die PINs dürften "falsch" in der platform.io hinterlegt sein. Sobald man GPIO8/9 verwendet, hat das einen Einfluss auf den Startup Mode (siehe https://docs.rs-online.com/2e8c/A700000008521020.pdf Seite 7 unten). Eine Konfiguration mit anderen Pins habe ich noch nicht getestet. Ist GPIO8/9 nämlich verbinden, bootet der ESP nicht mehr und das rote LED leuchtet. Ich werde es noch mit anderen GPIOs testen.

Hat es schon jemand erfolgreich getestet?

**Reply by fhempy** [reply]
Nach Änderung der PINs bootet der ESP zwar, aber NRF bleibt weiterhin auf "not connected" stehen. Habe noch weitere PINs ohne Erfolg getestet.
